### PR TITLE
Görli contracts deployment

### PR DESCRIPTION
**Pull request by nkuba**
Görli became a recommended test network after Ropsten's deprecation notice ([link](https://blog.ethereum.org/2022/06/21/testnet-deprecation/)).

In this PR we update deployment scripts and configuration to support Görli.

The contracts were already deployed to Görli:
`@keep-network/random-beacon@2.0.0-goerli.0`
`@keep-network/ecdsa@2.0.0-goerli.0`

### CODE REVIEW

File: solidity/random-beacon/hardhat.config.ts [hunk 1]
```diff
@@ -112,28 +112,32 @@ const config: HardhatUserConfig = {
...
+                "node_modules/@threshold-network/solidity-contracts/export/artifacts",
+              deploy:
+                "node_modules/@threshold-network/solidity-contracts/export/deploy",
+            },
+          ]
+        : undefined,
+    deployments: {
+      // For hardhat environment we can fork the mainnet, so we need to point it
+      // to the contract artifacts.
+      hardhat: process.env.FORKING_URL ? ["./external/mainnet"] : [],
```
Comment (michalinacienciala) [inline solidity/random-beacon/hardhat.config.ts:134]: Shouldn't we have some contracts (`T`?) in `solidity/random-beacon/external/mainnet/` for this to work?

**Review by michalinacienciala — CHANGES_REQUESTED** [review]
The deployment on hardhat network is failing with following error:
```
Error: ERROR processing /Users/michalina/workspace2/keep-core/solidity/random-beacon/deploy/02_deploy_beacon_sortition_pool.ts:
Error: No deployment found for: T
```

I think this may be related to the thing that I mentioned in the comment.

**Comment by michalinacienciala** [comment]
The checks didn't start after the force-push, but I checked the changes locally. The deploy on `goerli` has worked ok, but deploy on `hardhat` still fails with `No deployment found for: T` error.

File: solidity/random-beacon/hardhat.config.ts (hunk 1 above)
Comment (michalinacienciala) [reply solidity/random-beacon/hardhat.config.ts:134]: Ah, right... Then I think we should change the command we execute in the CI:
https://github.com/keep-network/keep-core/blob/77d7425e57e543581d069a5060f4ebcd3eec6dc0/.github/workflows/contracts-random-beacon.yml#L160

**Review by michalinacienciala — CHANGES_REQUESTED** [review]
We need to fix 2 things:
1. The deployment on `goerli` network now fails with `ProviderError: App is inactive. Please create a new app or contact customer support`. I tested this both locally and via GH Actions. This is the first time I see this error.
2. Command for deploying the contracts on `hardhat` env in GH Actions needs to be updated (see [comment](https://github.com/keep-network/keep-core/pull/3061#discussion_r930886778))

**Comment by nkuba** [comment]
>The deployment on goerli network now fails with ProviderError: App is inactive. Please create a new app or contact customer support. I tested this both locally and via GH Actions. This is the first time I see this error.

You need to use a new Alchemy API URL, as @pdyraga made some cleanup.

**Comment by michalinacienciala** [comment]
> > The deployment on goerli network now fails with ProviderError: App is inactive. Please create a new app or contact customer support. I tested this both locally and via GH Actions. This is the first time I see this error.
> 
> You need to use a new Alchemy API URL, as @pdyraga made some cleanup.

Ah, right, the workflow that I run from GH Actions used "old" env variables instead of `GOERLI_ETH_HOSTNAME_HTTP` and `GOERLI_ETH_CONTRACT_OWNER_PRIVATE_KEY`. And locally I used `goerli` Alchemy endpoint, but maybe it got deleted during @pdyraga's cleanup...

File: solidity/ecdsa/hardhat.config.ts [hunk 2]
```diff
@@ -87,23 +87,23 @@ const config: HardhatUserConfig = {
...
       chainId: 1101,
       tags: ["allowStubs"],
     },
-    ropsten: {
+    goerli: {
       url: process.env.CHAIN_API_URL || "",
-      chainId: 3,
+      chainId: 5,
       accounts: process.env.CONTRACT_OWNER_ACCOUNT_PRIVATE_KEY
         ? [process.env.CONTRACT_OWNER_ACCOUNT_PRIVATE_KEY]
```
Comment (michalinacienciala) [inline solidity/ecdsa/hardhat.config.ts:108]: Do we want to make similar change to [this one](https://github.com/keep-network/keep-core/pull/3111) here?

**Comment by michalinacienciala** [comment]
Now I'm getting:
```
[...]
transferring ownership of BeaconSortitionPool to 0x4CaAE804022Ba60bD3d28c1fC2f156eA340FDF36
hardhat-tenderly: No new contracts have been verified
approving the RandomBeacon in the TokenStaking
An unexpected error occurred:

Error: ERROR processing /home/runner/work/keep-core/keep-core/solidity/random-beacon/deploy/05_approve_random_beacon_in_token_staking.ts:
Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0x68ad60CC5e8f3B7cC53beaB321cf0e6036962dBc","to":"0xEDAA0fb826dB515031dE986Fc80EEEB035dd5281","data":"0xe3ae4d0a0000000000000000000000004caae804022ba60bd3d28c1fc2f156ea340fdf36","accessList":null}, error={"name":"ProviderError","code":-32000,"_isProviderError":true}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.6.8)
```

This occurs when I run the `random-beacon` contracts deployment workflow on `goerli` on a temporary branch that's combines the code from #3061 and #3081. I thought that maybe in the `random-beacon` we need the same change as was done in `keep-ecdsa` in https://github.com/keep-network/keep-core/pull/3095, but applying it didn't help... Workflow run: https://github.com/keep-network/keep-core/runs/7625056148?check_suite_focus=true.

**Comment by nkuba** [comment]
> 



> Now I'm getting:
> 
> ```
> [...]
> transferring ownership of BeaconSortitionPool to 0x4CaAE804022Ba60bD3d28c1fC2f156eA340FDF36
> hardhat-tenderly: No new contracts have been verified
> approving the RandomBeacon in the TokenStaking
> An unexpected error occurred:
> 
> Error: ERROR processing /home/runner/work/keep-core/keep-core/solidity/random-beacon/deploy/05_approve_random_beacon_in_token_staking.ts:
> Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0x68ad60CC5e8f3B7cC53beaB321cf0e6036962dBc","to":"0xEDAA0fb826dB515031dE986Fc80EEEB035dd5281","data":"0xe3ae4d0a0000000000000000000000004caae804022ba60bd3d28c1fc2f156ea340fdf36","accessList":null}, error={"name":"ProviderError","code":-32000,"_isProviderError":true}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.6.8)
> ```
> 
> This occurs when I run the `random-beacon` contracts deployment workflow on `goerli` on a temporary branch that's combines the code from #3061 and #3081. I thought that maybe in the `random-beacon` we need the same change as was done in `keep-ecdsa` in #3095, but applying it didn't help... Workflow run: https://github.com/keep-network/keep-core/runs/7625056148?check_suite_focus=true.

The `05_approve_random_beacon_in_token_staking.ts` script executes the transaction from `deployer` account:
https://github.com/keep-network/keep-core/blob/79121c789f0eb356e67b657659c7b1461c1d8690/solidity/random-beacon/deploy/05_approve_random_beacon_in_token_staking.ts#L13

This has to be the same account that deployed the `TokenStaking` contract in `@threshold-network/solidity-contracts` package.

File: solidity/random-beacon/hardhat.config.ts (hunk 1 above)
Comment (nkuba) [reply solidity/random-beacon/hardhat.config.ts:135]: 38ce9988d2f0e710452fb9845aebc7c5af6a8018

**Comment by michalinacienciala** [comment]
> Now I'm getting:
> 
> ```
> [...]
> transferring ownership of BeaconSortitionPool to 0x4CaAE804022Ba60bD3d28c1fC2f156eA340FDF36
> hardhat-tenderly: No new contracts have been verified
> approving the RandomBeacon in the TokenStaking
> An unexpected error occurred:
> 
> Error: ERROR processing /home/runner/work/keep-core/keep-core/solidity/random-beacon/deploy/05_approve_random_beacon_in_token_staking.ts:
> Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0x68ad60CC5e8f3B7cC53beaB321cf0e6036962dBc","to":"0xEDAA0fb826dB515031dE986Fc80EEEB035dd5281","data":"0xe3ae4d0a0000000000000000000000004caae804022ba60bd3d28c1fc2f156ea340fdf36","accessList":null}, error={"name":"ProviderError","code":-32000,"_isProviderError":true}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.6.8)
> ```

Sorry, that turned out to be a false alarm, it was my mistake with the `upstream_builds` configuration. I tested the `random-beacon` deployment on `hardhat` and `goerli` again and now deployment is passing.

**Review by michalinacienciala — APPROVED** [review]
I'm getting a failure when trying to deploy the `ecdsa` contracts on `goerli`, but we need the changes from https://github.com/keep-network/keep-core/pull/3096/files#diff-feafa53d66d92ad1dbcdd390448ac6b04423afcf7bd391d4e5b807dcb1c1dc15 for the successful deployment. So the plan is to merge #3061 and #3096 and do the final testing of the deployment as part of #3081.

**Pull request merged**